ionic-team/ionic-framework
# chore(vue): bump vue dev-deps to have generic type

Issue number: Internal

---------





## What is the current behavior?


In v3.2.46 [vue was updated](https://github.com/vuejs/core/pull/3969) to add support for a generic type argument to the `Plugin` type. In https://github.com/ionic-team/ionic-framework/pull/29637 Ionic's Vue plugin was updated to use the generic, but the`@ionic/vue` project is currently installing v3.2.37.

This results in a local type checking error and build error:
```
(!) Plugin typescript: @rollup/plugin-typescript TS2315: Type 'Plugin_2' is not generic.
src/ionic-vue.ts: (24:24)

24 export const IonicVue: Plugin<[IonicConfig?]> = {
```

## What is the new behavior?


- Bumps and pins the dev dependency of `vue` to `3.2.46`. There are other breaking changes that Vue has shipped in minor/patch cycles that prevents from updating to the latest. 
- Resolves the type checking error locally during build of the`@ionic/vue` package

## Does this introduce a breaking change?





## Other information



Validation steps:
1. Checkout the branch
2. Build `core/`
3. In `packages/vue`, install latest pinned dependencies with `npm ci`
4. Sync the changes to the `vue` package with `npm run sync`
5. Open `/packages/vue/src/ionic-vue.ts`
6. Observe: No type errors on L24
7. Run `npm run build`
8. Observe: No build errors
